update_initrd.sh

```diff
@@ -0,0 +1,43 @@
+#!/bin/bash
+
+set -e
+
+THISSCRIPT=$(basename $0)
+RULEFILES=$(find . -name initrd.rules)
+BUILDDIR="initrd"
+DEPFILE=".initrd.D"
+CPIOFILE="initrd.cpio"
+
+### TODO: It is possible to implement this script so that it would only copy the
+### modified files, and incrementally update the ramdisk. However, this is
+### difficult, and we don't keep thousands of files in the ramdisk, so
+### recreating entire cpio archive will be good for now.
+
+# Clean/create the build directory
+rm -rf $BUILDDIR
+mkdir -p $BUILDDIR
+
+SOURCEFILES=
+
+# Gather all files with rules, and copy files to the build dir
+for rulefile in $RULEFILES; do
+    echo "Processing file $rulefile"
+    sourcedir=$(dirname $rulefile)
+    rules=$(cat $rulefile)
+    while read -r line; do
+        source=$sourcedir/$(echo "$line" | cut  -d' ' -f1)
```

Comment:
I prefer to use following syntax to access words within a line:

```
$ cat blah
foo bar boo
$ var=($(<blah)); echo "$var[1]:$var[2]:$var[3]"
foo:bar:boo
```

Splitting line into words is controlled by `IFS` shell variable (Internal Field Separator).